include/zephyr/drivers/fuel_gauge.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+#define ZEPHYR_INCLUDE_DRIVERS_BATTERY_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/device.h>
+
+enum fuel_gauge_property_type {
+
+	/* Keep these alphabetized wrt property name */
+
+	/** Runtime Dynamic Battery Parameters */
+	/**
+	 * Provide a 1 minute average of the current on the battery.
+	 * Does not check for flags or whether those values are bad readings.
+	 * See driver instance header for details on implementation and
+	 * how the average is calculated. Units in mA negative=discharging
+	 */
+	FUEL_GAUGE_AVG_CURRENT = 0,
+
+	/** Battery current (mA); negative=discharging */
+	FUEL_GAUGE_CURRENT,
+	/** Whether the battery underlying the fuel-gauge is cut off from charge */
+	FUEL_GAUGE_CHARGE_CUTOFF,
+	/** Cycle count in 1/100ths (number of charge/discharge cycles) */
+	FUEL_GAUGE_CYCLE_COUNT,
+	/** Connect state of battery */
+	FUEL_GAUGE_CONNECT_STATE,
+	/** General Error/Runtime Flags */
+	FUEL_GAUGE_FLAGS,
+	/** Full Charge Capacity in mAh (might change in some implementations to determine wear) */
+	FUEL_GAUGE_FULL_CHARGE_CAPACITY,
+	/** Is the battery physically present */
+	FUEL_GAUGE_PRESENT_STATE,
+	/** Remaining capacity in mAh */
+	FUEL_GAUGE_REMAINING_CAPACITY,
+	/** Remaining battery life time in minutes */
+	FUEL_GAUGE_RUNTIME_TO_EMPTY,
+	/** Remaining time in minutes until battery reaches full charge */
+	FUEL_GAUGE_RUNTIME_TO_FULL,
+	/** Absolute state of charge (percent, 0-100) - expressed as % of design capacity */
+	FUEL_GAUGE_STATE_OF_CHARGE,
+	/** Temperature in 0.1 K */
+	FUEL_GAUGE_TEMPERATURE,
+	/** Battery voltage (mV) */
+	FUEL_GAUGE_VOLTAGE,
+
+	/** Reserved to demark end of common fuel gauge properties*/
+	FUEL_GAUGE_COMMON_COUNT,
+	/** Reserved to demark downstream custom properties */
+	FUEL_GAUGE_CUSTOM_BEGIN = FUEL_GAUGE_COMMON_COUNT,
+
+	/** Reserved to demark end of valid enum properties */
+	FUEL_GAUGE_PROP_MAX = UINT16_MAX
+};
```

Comment:
Verified, if you use C++ downstream users will not be able to do something like:

```
#define MY_PROPERTY FUEL_GAUGE_CUSTOM_BEGIN
```

And pass it as the enum, C++ is more strongly typed. It is possible to include the cast in the #define to look like:
```
#define MY_PROPERTY ((enum fuel_gauge_property_type)FUEL_GAUGE_CUSTOM_BEGIN)
```

But I think that at this point you're losing the type safety anyway. We should just use #define for all these values instead.

Reply:
Actually I played with this some more and I think we're far more likely to add properties here. Let's keep the enum

Reply:
Thanks for testing and pointing this out. I've converted the enum to `#define`s.